### DISCUSSION TITLE
PostConstruct Annotation Question

### DISCUSSION THREAD

**Discussion by GreihMurray** [micronaut-projects/micronaut-core#8798] [Q&A]
Can anyone help me understand why the PostConstruct annotation is meant to not execute if a bean is instantiated using new? For example, using "var bean = new Bean()" will not trigger the PostConstruct annotation but something like "@Inject Bean bean" will. What is the difference and why does only one trigger PostConstruct?

**Comment by graemerocher** [comment]
a directly instantiated object is considered un-managed. When you use `@Inject` the Micronaut container can manage the lifecycle of the object (including executing post construct and pre destroy hooks). When you directly instantiate you are on your own and Micronaut cannot help you.
